Improve documentation on setting a dev environment

julien4215 opened the pull request:
## Changes

- completed the android emulator troubleshooting section
- added the section on setting up a real device
- moved the When using a manually installed lila server and When using lila-docker sections that were in Setting up the emulators to the Run section
- changed the content of When using lila-docker
- moved some part of Local lila server (manual installation) to Run and also small changes to Run

I also changed the scheme when using a private ip address for the host so that you can connect to a local lila from your android phone without using `adb reverse`.

veloce commented:
Can you summarise quickly what are the changes here?

Basically what did you remove, what did you add in this doc? thank.

julien4215 commented:
- completed the android emulator troubleshooting section
 - added the section on setting up a real device
 - moved the **When using a manually installed lila server** and **When using lila-docker** sections that were in **Setting up the emulators** to the **Run** section
 - changed the content of **When using lila-docker**
 - moved some part of **Local lila server (manual installation)** to **Run** and also small changes to **Run**
